packages/vx-xychart/src/types/data.ts

```diff
@@ -0,0 +1,66 @@
+import { AxisScaleOutput } from '@vx/axis';
+import {
+  ScaleTypeToD3Scale,
+  ScaleConfig,
+  ScaleConfigToD3Scale,
+  ScaleInput,
+  D3Scale,
+} from '@vx/scale';
+import DataRegistry from '../classes/DataRegistry';
+import { XYChartTheme } from './theme';
+
+export type Margin = {
+  top: number;
+  right: number;
+  bottom: number;
+  left: number;
+};
+
+export type LegendShape = 'rect' | 'line' | 'dashed-line' | 'circle';
+
+export interface DataRegistryEntry<
+  XScaleConfig extends ScaleConfig<AxisScaleOutput>,
```

Comment:
Since `XScaleConfig` and `YScaleConfig` are not used directly, accepting generics `XScale` and `YScale` instead might be simpler. 

```ts
type AxisScale = D3Scale<AxisScaleOutput>; // not sure if `,any,any` is needed.

export interface DataRegistryEntry<
  XScale extends AxisScale,
  YScale extends AxisScale,
  Datum
> {
  /** unique data key */
  key: string;
  /** array of data for the key. */
  data: Datum[];
  /** function that returns the x value of a datum. */
  xAccessor: (d: Datum) => ScaleInput<XScale>;
  /** function that returns the y value of a datum. */
  yAccessor: (d: Datum) => ScaleInput<YScale>;
  /** whether the entry supports mouse events. */
  mouseEvents?: boolean;
  /** Optionally update the xScale. */
  xScale?: (xScale: XScale) => XScale;
  /** Optionally update the yScale. */
  yScale?: (yScale: YScale) => YScale;
  /** Legend shape for the data key. */
  legendShape?: LegendShape;
}
```